valhalla/baldr/conditional_speed_limit.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+namespace valhalla {
+namespace baldr {
+
+struct ConditionalSpeedLimit {
+  /**
+   * Constructs `ConditionalSpeedLimit` from the provided speed and condition.
+   * @param speed Speed limit value in KPH
+   * @param condition `TimeDomain` value of the condition. Can be accuired from the
+   * `TimeDomain::td_value()` or from `get_time_range(opening_hours_str)`.
+   */
+  explicit ConditionalSpeedLimit(uint8_t speed, uint64_t condition);
+
+  ConditionalSpeedLimit() : ConditionalSpeedLimit(0, 0) {
+  }
+
+  /**
+   * Provides speed limit value in KPH.
+   * @return Speed limit value in KPH.
+   */
+  uint32_t speed_limit() const {
+    return speed_limit_;
+  }
+
+  /**
+   * Provides the `TimeDomain` value of the condition, that defines when this speed limit is applied.
+   * @return `TimeDomain` value of the condition.
+   */
+  uint64_t condition() const;
+
+  /**
+   * String representation of the condition of this speed limit. Intended to use for debug purpose.
+   * @return Condition string representation in the opening_hours format.
+   */
+  std::string condition_str() const;
+
+private:
+  // This structure basically copies `TimeDomain` definition and adds speed_limit value into the spare
+  // bits, which allows us to use only 8 bytes per conditional speed limit.
+  uint64_t day_dow_type_ : 1;  // type of day_dow, 0 - day of month, 1 - nth day of week
+  uint64_t dow_ : 7;           // day of week mask for this restriction
+  uint64_t begin_hrs_ : 5;     // begin hours
+  uint64_t begin_mins_ : 6;    // begin minutes
+  uint64_t begin_month_ : 4;   // begin month
+  uint64_t begin_day_dow_ : 5; // begin day of month or nth dow, i.e. 1st Sunday
+  uint64_t begin_week_ : 3;    // which week does this start, i.e. 1st week in Oct
+  uint64_t end_hrs_ : 5;       // end hours
+  uint64_t end_mins_ : 6;      // end minutes
+  uint64_t end_month_ : 4;     // end month
+  uint64_t end_day_dow_ : 5;   // end day of month or nth dow, i.e. last Sunday
+  uint64_t end_week_ : 3;      // which week does this end, i.e. last week in Oct
+  uint64_t spare_ : 2;
+
+  uint64_t speed_limit_ : 8; // speed limit in KPH
```

Comment:
Basically `TimeDomain::spare_` are used here to store the speed limit. Alternatively the bit mask can be used to write speed limit on top of the `TimeDomain` value with cost of the implicit dependency if one day someone decides to use `TimeDomain::spare_` for something, so I decided to go with copy&paste here.